CJBCheatsMenu/Framework/Cheats/FarmAndFishing/NoAnimalFriendshipDecayCheat.cs

```diff
@@ -0,0 +1,57 @@
+using System.Collections.Generic;
+using System.Linq;
+using CJBCheatsMenu.Framework.Components;
+using StardewModdingAPI;
+using StardewModdingAPI.Events;
+using StardewValley;
+using StardewValley.Menus;
+
+namespace CJBCheatsMenu.Framework.Cheats.FarmAndFishing
+{
+    /// <summary>A cheat which prevents NPC friendships from decaying.</summary>
+    internal class NoAnimalFriendshipDecayCheat : BaseCheat
+    {
+        /*********
+        ** Fields
+        *********/
+        /*********
+        ** Public methods
+        *********/
+        /// <summary>Get the config UI fields to show in the cheats menu.</summary>
+        /// <param name="context">The cheat context.</param>
+        public override IEnumerable<OptionsElement> GetFields(CheatContext context)
+        {
+            yield return new CheatsOptionsCheckbox(
+                label: I18n.Farm_NoDecay(),
+                value: context.Config.NoAnimalFriendshipDecay,
+                setValue: value => context.Config.NoAnimalFriendshipDecay = value
+            );
+        }
+
+        /// <summary>Handle the cheat options being loaded or changed.</summary>
+        /// <param name="context">The cheat context.</param>
+        /// <param name="needsUpdate">Whether the cheat should be notified of game updates.</param>
+        /// <param name="needsInput">Whether the cheat should be notified of button presses.</param>
+        /// <param name="needsRendering">Whether the cheat should be notified of render ticks.</param>
+        public override void OnConfig(CheatContext context, out bool needsInput, out bool needsUpdate,
+            out bool needsRendering)
+        {
+            needsInput = false;
+            needsUpdate = context.Config.NoAnimalFriendshipDecay;
+            needsRendering = false;
+        }
+
+        /// <summary>Handle the player loading a save file.</summary>
+        /// <param name="context">The cheat context.</param>
+        /// <summary>Handle a game update if <see cref="ICheat.OnSaveLoaded"/> indicated updates were needed.</summary>
+        /// <param name="context">The cheat context.</param>
+        /// <param name="e">The update event arguments.</param>
+        public override void OnUpdated(CheatContext context, UpdateTickedEventArgs e)
+        {
+            if (!e.IsOneSecond || !Context.IsWorldReady)
+                return;
+            IEnumerable<FarmAnimal> animals = Game1.getFarm().getAllFarmAnimals().Distinct();
+            foreach (FarmAnimal animal in animals) animal.wasPet.Set(true);
```

Comment:
We should indent blocks for consistency, and we can use the more conventional `Value` property:

```cs
foreach (FarmAnimal animal in animals)
    animal.wasPet.Value = true;
```